### DISCUSSION TITLE
Multiple template files

### DISCUSSION THREAD

**Discussion by RobinManoli** [pinecone-router/router#44] [Ideas]
Hi there, thanks for the awesome router!
Since you're having the option of multiple handlers within the x-handler, why not multiple template files?

My use case it that I want to keep html code in different files, and I want to include more than one of these files on a specific path. Since the x-template attribute is so convenient, it would be nice to add all files there.

Is there any way to do this out of the box with javascript, in the handler code?

**Comment by rehhouari** [comment]
Hello, sorry for the very late reply!
I will try to look into implementing this right now.
I can't think of an easy way to do this with a handler, you'd have to make your own list of routes and their templates somewhere then access it. it's easy but defeats the purpose kinda, since you'd be repeating the route definitions.. unless you only ever use [programmatic route definitions](https://github.com/pinecone-router/router?tab=readme-ov-file#adding-and-removing-routes--templates-programmatically-with-javascript) which again kinda defeat the purpose of alpine-specific router.

This sounds like a good idea tho thanks!

**Comment by rehhouari** [comment]
I just pushed 5.0.0 update with this feature!
Hopefully its not too late and it works well
